lib/index.ts

```diff
@@ -225,7 +225,10 @@ export default class Doshii {
         },
       });
       return resp.data;
-    } catch (error) {
+    } catch (error: any) {
+      if (axios.isAxiosError(error)) {
+        delete error.config?.headers?.authorization;
```

Comment:
Considering this is a logging concern, should we not be updating the logging function with Smart Hospo to redact this. The `pino` module that is being used has the capability to define properties that should be redacted, without writing any explicit log manipulation.

e.g.

```
redact: {
      paths: ['metadata.headers.authorization']
}
```

The concern with this solution in the partner SDK is that we are assuming other parties using this SDK don't want the authorization header in the axios header, which I think is an incorrect assumption.

Plus, I'm also not sure by deleting the `authorization` property from the error object, will that actually have a flow-on effect elsewhere? We might actually be deleting the property from a global object that is being used in subsequent calls, resulting in potential errors. I'm not sure if axios uses immutable concepts, so it's possible there could be flow-on effects of doing this property deletion. 🤔 

My recommendation would be to update the pino configuration in Smart Hospo to redact specific properties.

https://getpino.io/#/docs/redaction

Reply:
Hey @johnatdoshii, exposing the token in the logs is a security issue for all partners, isn't it? that's why I asked Jack to fix this in the SDK.

Reply:
@carlosatdoshii this code is not exposing the token in the logs. The error is thrown. It is the caller that is deciding whether or not to log. We don't know what the caller will be doing, it could catch the error and retry (without logging), using the same details from the error. The logging is being done by Smart Hospo, not this SDK. 🤔